libntfs-3g/index.c

```diff
@@ -578,6 +578,17 @@ int ntfs_index_entry_inconsistent(ntfs_volume *vol, INDEX_ENTRY *ie,
 	if (!ie)
 		return 0;
 
+	if (ie->length == 0 && ie->key_length == 0) {
+		check_failed("Index entry length is zero, It should be at least INDEX_ENTRY_HEADER(16) size");
+		if (ntfsck_ask_repair(vol)) {
+			ie->length = cpu_to_le16(sizeof(INDEX_ENTRY_HEADER));
+			ie->ie_flags = INDEX_ENTRY_END;
+			ret = 1;
+			fsck_err_fixed();
+		} else
+			ret = -1;
```

Comment:
wouldn't be nice to exit here? there's no meaning to check condition below if ie length is zero?
and I think other cases also are nice to exit there.
Or, is there any cases to change return value not zero to zero? if isn't, I think it's better to exit return value is confirmed.

Reply:
Done. thanks